src/domain/schema/keywords/is-date.keyword.ts

```diff
@@ -0,0 +1,37 @@
+import Ajv from 'ajv';
+
+/**
+ * Registers isDate keyword in AJV
+ *
+ * When the value of isDate is set to true, the field will be validated
+ * according to ISO 8601.
+ *
+ * If the date is valid, the target property is created as {@link Date}.
+ * If the date is invalid, a validation error is thrown.
+ *
+ * Null date values are supported – we assume that a null date is a valid
+ * date.
+ *
+ * @param ajv - the AJV instance to which the isDate keyword should be registered
+ */
+export function addIsDate(ajv: Ajv) {
+  ajv.addKeyword({
+    keyword: 'isDate',
+    compile: (schema) => (value, obj) => {
```

Comment:
I'd give `obj` a more meaningful name. I think it refers to the parent schema, so maybe `parentSchema` is a good name, as it is [in the docs](https://ajv.js.org/keywords.html#define-keyword-with-compile-function) 🙂 Wdyt?

Reply:
Maybe `data` and `dataContext`?

See: https://github.com/ajv-validator/ajv/blob/948ffbf07345eeca286e50cc233bedc3f72e8f03/lib/types/index.ts#L51

Reply:
Sure, I'm ok with it 🙂